### PR TITLE
Allow accounts with null ID to enable ADALv3 migration

### PR DESCRIPTION
**Pull request by bgavrilMS**

### CODE REVIEW

**Comment by bgavrilMS** [comment]
Note that I am still looking at writing a unit test, but this should unblock E2E testing

**Review by jennyf19 — APPROVED** [review]
:shipit: Nice catch

File: msal/src/Microsoft.Identity.Client/IAccount.cs [hunk 1]
```diff
@@ -53,7 +53,7 @@ public interface IAccount
         /// <summary>
         /// AccountId of the home account for the user. This uniquely identifies the user across AAD tenants. 
         /// </summary>
-        /// <remarks>Can be null</remarks>
+        /// <remarks>Can be null, for example with users migrating from ADAL v3</remarks>
```
Comment (jmprieur) [inline msal/src/Microsoft.Identity.Client/IAccount.cs]: Can be null, for example if this account was migrated to MSAL.NET from ADAL.NET v3's token cache

**Review by jmprieur — APPROVED** [review]
LGTM. 
proposed a slight change in the XML comment

**Review by henrik-me — APPROVED** [review]
:shipit:

**Pull request merged**